### DISCUSSION TITLE
Support 3D geometries

### DISCUSSION THREAD

**Discussion by kevinschaich** [Turfjs/turf#2792] [Feature proposals]
Is it possible to support 3D geometries with an optional `z` / `altitude` / `elevation` coordinate? 

This is now supported by the [GeoJSON spec](https://datatracker.ietf.org/doc/rfc7946/).

**Comment by smallsaucepan** [comment]
Thanks for raising this @kevinschaich. To what extent are you requesting support? Let's say the spectrum runs from "let people provide z coords but ignore them for calculations" all the way up to "fully incorporate altitude variations in to all functionality".

Do you have a particular use case? Or are you getting errors when supplying a z coord?

**Comment by pvy** [comment]
E.g. bezierSpline ignores the z coordinate and returns two-dimensional coordinates.

Other 2D functions could project points onto a plane and work inside it.

**Comment by kevinschaich** [comment]
Ignore z coordinate would be a great first step.

I imagine things get much more complicated if you're doing the actual operations – [the GeoJSON spec](https://www.rfc-editor.org/rfc/rfc7946#section-4) says:

> An OPTIONAL third-position element SHALL be the height in meters above or below the WGS 84 reference ellipsoid.

so that would mean the three coordinates are not in the same units (`[lon <WGS84 decimal degrees>, lat <WGS84 decimal degrees>, alt <meters>]`).

Do all the functions share a common parser step at the beginning? If so, would be easy to just strip off the third coordinate and then pass that in.

**Reply by pbelbin** [reply]
anything happening here?  3D distance support would be great to have.  ie: path from a to b is seldom actually on a flat surface, and variations mean an increase in distance travelled.  imagine someone taking a personal distance measuring device, and they climb and descend hills/mountains.  they've travelled a lot further than what a flat, straight line between the two sides of a hill/mountain would suggest.
